Assets/MixedRealityToolkit.Examples/Demos/EyeTracking/Demo_Visualizer/Scripts/DrawOnTexture.cs

```diff
@@ -0,0 +1,364 @@
+    {
+        public Texture2D HeatmapLookUpTable; //TODO: Replace with gradient?
+        //public Color drawBrushColor;
+        public float drawBrushSize; // aka spread
+        public float drawIntensity; // aka amplitude
+        public float minThreshDeltaHeatMap = 0.001f; // Mostly for performance to reduce spreading heatmap for small values.
+        public bool UseLiveInputStream = false;
+
+        private Texture2D myDrawTex;
+        private Renderer myRenderer;
+
+        public Material HeatmapOverlayMaterialTemplate;
+
+        private void Update()
+        {
+            if (UseLiveInputStream && MixedRealityToolkit.InputSystem?.EyeGazeProvider?.GazeTarget == gameObject && MixedRealityToolkit.InputSystem.EyeGazeProvider.IsEyeGazeValid)
+            {
+                DrawAtThisHitPos(MixedRealityToolkit.InputSystem.EyeGazeProvider.HitPosition);
+            }
+        }
+
+        public void DrawAtThisHitPos(Vector3 hitPosition)
+        {
+            Vector2? hitPosUV = GetCursorPosInTexture(hitPosition);
+            if (hitPosUV != null)
+            {
+                StartCoroutine(DrawAt(hitPosUV.Value));
+            }
+        }
+
+        private void DrawAt(Vector2 posUV, Color col)
+        {
+            if (MyDrawTexture != null)
+            {
+                Debug.LogFormat("New draw point at ( {0}; {1} )", posUV.x, posUV.y);
+                MyDrawTexture.SetPixel(
+                    (int)(posUV.x * MyDrawTexture.width), 
+                    (int)(posUV.y * MyDrawTexture.height),
+                    col);
+
+                MyDrawTexture.Apply();
+            }
+        }
+
+
+        bool neverDrawnOn = true;
+        Vector2 prevPos = new Vector2(-1, -1);
+        float dynamicRadius = 0;
+
+        private void DrawAt2(Vector2 posUV, int maxRadius, float intensity)
+        {
+            //TODO:
+            // * intensity: More efficient data type? should only be between 0 and 1
+            //
+            if (MyDrawTexture != null)
+            {
+                // Reset on first draw
+                if (neverDrawnOn)
+                {
+                    for (int ix = 0; ix < MyDrawTexture.width; ix++)
+                    {
+                        for (int iy = 0; iy < MyDrawTexture.height; iy++)
+                        {
+                            MyDrawTexture.SetPixel((int)(ix), (int)(iy), new Color(0, 0, 0, 0));   
+                        }
+                    }
+                    neverDrawnOn = false;
+                }
+
+                // Determine the center of our to be drawn 'blob'
+                Vector2 center = new Vector2(posUV.x * MyDrawTexture.width, posUV.y * MyDrawTexture.height);
+
+                // Determine appropriate radius based on viewing duration: The longer looking in a small region the larger the radius
+                float dist = maxRadius;
+                //float dist = Vector2.Distance(center, prevPos);
```

Comment:
This file needs a commented code scrub. A lot of things below should be removed

Reply:
Scrubbed this file.